---
title: "Fungi_Spermosphere_Preprocessing"
author: "Emily Roggenkamp"
date: "2025-06-13"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r Loading Libraries and Pallettes, warning=FALSE, message=FALSE}
##Loading Libraries ##

library(phyloseq)
library(decontam)
library(vegan)
library(tidyverse)
library(metagenomeSeq)
library(ggplot2)
library(ggpubr)
library(Biostrings)
library(microbiome)

##### Set global options #####
# no scientific notation
options(scipen=10000) 
#Since RMD files moved the working directories needed to be specified

# color blind pallets used throughout 
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ibm.cbb <- c("#648FFF", "#785EF0", "#DC267F", "#FE6100", "#FFB000")
tol.cbb <- c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255")

# Extra colors 
library(RColorBrewer)
n <- 40
qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
```

## Input Files
```{r}
# Metadata #
samp_dat_fun <- read.csv("Spermosphere_ITS_Metadata.csv", na.strings = "na")

rownames(samp_dat_fun) <- samp_dat_fun$Sample_ID #row names must match OTU table headers
SAMP.fun <- phyloseq::sample_data(samp_dat_fun)

#OTU table # remove hash and space in OTU_ID header manually
otu_fun <- read.csv("Usearch_Merged_Analysis/otu_table_ITS_Fungi.txt", sep = "\t") #txt file is tab separated
colnames(otu_fun) <- sub("_.*", "", colnames(otu_fun)) #removing _S*_merged
rownames(otu_fun) <- otu_fun$OTU
otu_fun <- otu_fun[,-1]
OTU.fun <- phyloseq::otu_table(otu_fun, taxa_are_rows = TRUE)
any(is.na(otu_fun)) # no NA in the OTU table
###IN ORDER TO KNIT THIS THE LINES ABOVE ARE COMMENTED OUT FILE IS COMPRESSED ###
# Taxonomy #
taxonomy.fun <- read.csv("Usearch_Merged_Analysis/constax_taxonomy.txt", header = TRUE, sep ="\t")
taxonomy.fun[, "Kingdom"] <- as.factor(gsub("_1", "", taxonomy.fun[, "Kingdom"]))
taxonomy.fun[, "Phylum"] <- as.factor(gsub("_1", "", taxonomy.fun[, "Phylum"]))
taxonomy.fun[, "Class"] <- as.factor(gsub("_1", "", taxonomy.fun[, "Class"]))
taxonomy.fun[, "Order"] <- as.factor(gsub("_1", "", taxonomy.fun[, "Order"]))
taxonomy.fun[, "Family"] <- as.factor(gsub("_1", "", taxonomy.fun[, "Family"]))
taxonomy.fun[, "Genus"] <- as.factor(gsub("_1", "", taxonomy.fun[, "Genus"]))
taxonomy.fun[, "Species"] <- as.factor(gsub("_1", "", taxonomy.fun[, "Species"]))

rownames(taxonomy.fun) <- taxonomy.fun$OTU_ID
TAX.fun <- phyloseq::tax_table(as.matrix(taxonomy.fun))


FASTA.fun <- readDNAStringSet("Usearch_Merged_Analysis/otus.fasta", format="fasta", seek.first.rec=TRUE, use.names=TRUE)

###### Create Initial Phyloseq object #####
# Merge reads into Phyloseq object #
fun.for <- phyloseq::phyloseq(OTU.fun, TAX.fun, SAMP.fun, FASTA.fun)
###IN ORDER TO KNIT THIS THE LINE ABOVE IS COMMENTED OUT; MULTIPLE FILES ARE COMPRESSED ###
#save objects to reduce memory?
#saveRDS(fun.for, file = "Spermosphere_Fungi_Full.rds")
#fun.for <- readRDS(file = "Fungi_RDS/Spermosphere_Fungi_Full.rds")
```

```{r Magbio Bead Normalization Test}
#samp_dat_mag <- read.csv("Magbead_Metadata.csv", na.strings = "na")
#rownames(samp_dat_mag) <- samp_dat_mag$Sample_ID
#SAMP.mag <- phyloseq::sample_data(samp_dat_mag)

#mag.fun.for <- phyloseq::phyloseq(OTU.fun, TAX.fun, SAMP.mag, FASTA.fun)
#saveRDS(mag.fun.for, file = "magbead_physeq_fun.rds")
```

# Decontaminate ##
```{r Decontamination}
unique(fun.for@sam_data$Sample_Type)

sample_data(fun.for)$is.neg <- sample_data(fun.for)$Sample_Type == "Extraction_Control" | sample_data(fun.for)$Sample_Type == "PCR_Control"
contamdf.prev <- isContaminant(fun.for, method="prevalence", neg="is.neg", threshold = 0.1, normalize = TRUE)
badTaxa <- rownames(contamdf.prev[contamdf.prev$contaminant == TRUE,])

print(badTaxa)

ps.pa <- transform_sample_counts(fun.for, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$is.neg == "TRUE", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$is.neg == "FALSE", ps.pa)

# Make data.frame of prevalence in positive and negative samples
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prev$contaminant)
decontaminate.fun <- ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + 
  geom_point() +
  xlab("Prevalence (Negative Controls)") + 
  ylab("Prevalence (True Samples)") + 
  scale_color_manual(values = cbbPalette)+ 
  ggtitle("Fungi") +
  theme_classic()
goodTaxa <- setdiff(taxa_names(fun.for), badTaxa)
str(goodTaxa) 
fun_f_filt <- prune_taxa(goodTaxa, fun.for)
```

Using Usearch merged reads since it merged ~90% of the reads.

Usearch Merged Reads Decontamination:
17 bad taxa
26355 good taxa


# Taxonomy filtering #
```{r Filtering Taxa that are not Bacteria}
# remove OTUs that are not Fungi 
fun_filt <- fun_f_filt %>% 
  phyloseq::subset_taxa(Kingdom == "Fungi")

fun_filt 
```

# Mock Community analysis #
```{r Mock Community Analysis}
# positive controls
fun_mock <- fun.for %>% 
  subset_samples(Sample_Type == "Positive_Control") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 1, TRUE) #%>%
  #phyloseq::filter_taxa(function(x) sum(x) > 1500, TRUE) # filter OTUs to have more than 10 read in mock samples

mock2 <- microbiome::transform(fun_mock, "compositional") # relative abundance transform

sequenced.mock.fun <- mock2 %>%
  phyloseq::filter_taxa(function(x) sum(x) > .005, TRUE) %>%
  psmelt() %>% 
  ggplot(aes(Sample, Abundance, fill = OTU)) +
  geom_bar(stat = "identity") +
  theme_classic() +
  scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  labs(x = "", y = "Relative abundance (%)",
       title = "Fungi") + 
  theme(axis.text.x = element_text(angle=45, hjust=1)) 
sequenced.mock.fun

#Add theoretical mock community?

mock.composition <- mock2 %>%
  psmelt() %>%
  dplyr::group_by(OTU) %>%
  dplyr::summarise(MeanRelAbund = mean(Abundance)) %>%
  arrange(-MeanRelAbund)

# these 8 OTUs made up 99.3% of the mock composition. These OTUs also match the 8 supposed to be in the mock
sum(mock.composition[1:8,]$MeanRelAbund)
```

# Data filtering #
```{r Removing Positve and Negative Controls}
# remove samples with less than 5000 reads
fun_sperm <- fun_filt %>% 
  subset_samples(Sample_Type %in% c("Bulk_Soil", "Spermosphere")) #%>%
  #prune_samples(sample_sums(.) > 5000, .) %>% # remove samples below 5000 reads
  #phyloseq::filter_taxa(function(x) sum(x) > 1, TRUE) # remove taxa with less than 2 reads
sample.sums <- data.frame(sample_sums(fun_sperm))

###### RDS of Non-normalized Fungal data ######
# Save an object to a file
#saveRDS(fun_sperm, file = "Spermosphere_Fungi_clean.rds")
```

# Read Depth Analysis
```{r Read Depth Analysis}
# Restore the object
#bac_sperm <- readRDS(file = "Bacteria_RDS/Spermosphere_bac_clean.rds")

###### READS PER SAMPLE ######
sample.sums <- data.frame(sample_sums(fun_sperm))

view(sample.sums)

read.dist.fun <- ggplot(sample.sums, aes(x = sample_sums.fun_sperm.)) +
  geom_histogram(color = "black", fill = cbbPalette[[4]]) + 
  theme_classic() +
  xlab("Read Depth") + 
  ggtitle("Fungi")

sum(sample_sums(fun_sperm)) 
median(sample_sums(fun_sperm)) 
str(fun_sperm@sam_data) 
str(fun_sperm@otu_table) 

```

Usearch merged:
52,348,646 total reads
133,953 median reads per sample
384 samples
11,701 OTUs



# Rarefaction Analysis #
```{r Rarefaction Analysis}
###### Rarefaction analysis #####
sam.data <- data.frame(fun_sperm@sam_data)
fOTU.table <- otu_table(fun_sperm) %>%
  as.data.frame() %>%
  as.matrix()

raremax <- min(rowSums(t(fOTU.table)))
raremax2 <- min(rowSums(fOTU.table))

rare.fun <- rarecurve(t(fOTU.table), step = 1000, sample = raremax, tidy = T)

fun.rare.curve.extract2 <- left_join(sam.data, rare.fun, by = c("Sample_ID" = "Site"))

fun.rare <- ggplot(fun.rare.curve.extract2, aes(x = Sample, y = Species, group = Sample_ID, color = Sample_Type)) + 
  #geom_point() +
  geom_line() + 
  xlab("Reads") + 
  ylab("Number of OTUs") +
  ggtitle("Fungi") +
  theme_classic() + 
  geom_vline(xintercept = median(sample_sums(fun_sperm)), linetype = "dashed") +
  scale_color_manual(values = cbbPalette)

p1 <- ggarrange(fun.rare, 
          read.dist.fun, 
          sequenced.mock.fun, 
          decontaminate.fun, nrow = 2, ncol = 2, labels = "auto")

p1

#ggsave(filename = "preprocess_usearch25.png", plot = p1, device = "png", width = 600, height = 400, units="mm", dpi = 600)
```

# Metagenome CSS normalization #
```{r Metagenome CSS normalization and Rarefying}
MGS <- phyloseq_to_metagenomeSeq(fun_sperm) #converts to metagenomeseq format
p <- metagenomeSeq::cumNormStatFast(MGS)
MGS <- metagenomeSeq::cumNorm(MGS, p =p)
metagenomeSeq::normFactors(MGS) # exports the normalized factors for each sample
norm.fun <- metagenomeSeq::MRcounts(MGS, norm = T) 
norm.fun.OTU <- phyloseq::otu_table(norm.fun, taxa_are_rows = TRUE) #exports the new otu table

fun.css.norm <- phyloseq::phyloseq(norm.fun.OTU, FASTA.fun, SAMP.fun, TAX.fun) #new otu table phyloseq object
# ABOVE LINE IS COMMENTED OUT BECAUSE THE FILE IS ZIPPED IN A FILE #
#saveRDS(fun.css.norm, file = "Fungi_RDS/Spermosphere_Fungi_CSS.rds")
# ABOVE LINE IS COMMENTED OUT BECAUSE THE FILE IS ZIPPED IN A FILE #

# Restore the object
#fun.css.norm <- readRDS(file = "Bacteria_RDS/Spermosphere_CSS.rds")
```

# Further Filtering #

As we are only interested in what's most abundant, we filtered further.

```{r}
fun_filter <- genefilter_sample(fun.css.norm, 
                                filterfun_sample(function(x) x > 5),
                                A=.03*nsamples(fun.css.norm))
fun.pruned <- prune_taxa(fun_filter, fun.css.norm)
```
# Reformat Taxonomy #

Using Nico's code to get highest classified level and "FOTU_taxa" label
Add the following functions. This is for constax taxonomy! dada2 taxonomy, we add the label column before loading into phyloseq object

```{r Reformat Taxonomy Functions}
blank2na = function(x, na.strings=c('','.','NA','na','N/A','n/a','NaN','nan')) {
  if (is.factor(x)) {
    lab = attr(x, 'label', exact = T)
    labs1 <- attr(x, 'labels', exact = T)
    labs2 <- attr(x, 'value.labels', exact = T)
    # trimws will convert factor to character
    x = trimws(x,'both')
    if (! is.null(lab)) lab = trimws(lab,'both')
    if (! is.null(labs1)) labs1 = trimws(labs1,'both')
    if (! is.null(labs2)) labs2 = trimws(labs2,'both')
    if (!is.null(na.strings)) {
      # convert to NA
      x[x %in% na.strings] = NA
      # also remember to remove na.strings from value labels 
      labs1 = labs1[! labs1 %in% na.strings]
      labs2 = labs2[! labs2 %in% na.strings]
    }
    # the levels will be reset here
    x = factor(x)
    if (! is.null(lab)) attr(x, 'label') <- lab
    if (! is.null(labs1)) attr(x, 'labels') <- labs1
    if (! is.null(labs2)) attr(x, 'value.labels') <- labs2
  } else if (is.character(x)) {
    lab = attr(x, 'label', exact = T)
    labs1 <- attr(x, 'labels', exact = T)
    labs2 <- attr(x, 'value.labels', exact = T)
    # trimws will convert factor to character
    x = trimws(x,'both')
    if (! is.null(lab)) lab = trimws(lab,'both')
    if (! is.null(labs1)) labs1 = trimws(labs1,'both')
    if (! is.null(labs2)) labs2 = trimws(labs2,'both')
    if (!is.null(na.strings)) {
      # convert to NA
      x[x %in% na.strings] = NA
      # also remember to remove na.strings from value labels 
      labs1 = labs1[! labs1 %in% na.strings]
      labs2 = labs2[! labs2 %in% na.strings]
    }
    if (! is.null(lab)) attr(x, 'label') <- lab
    if (! is.null(labs1)) attr(x, 'labels') <- labs1
    if (! is.null(labs2)) attr(x, 'value.labels') <- labs2
  } else {
    x = x
  }
  return(x)
}

# In the tax_table add a column naming the highest resolution taxonomy 
# achieved for each OTU, remove _ and add sp. to genera
ReformatTaxonomy <- function(dataframe){
  taxa_table = as(tax_table(dataframe), "matrix")
  taxa_table = as.data.frame(taxa_table)
  # remember to do run this function only once on your dataframe
  taxa_table$Genus <- as.character(taxa_table$Genus)
  taxa_table[taxa_table=="Unclassified"]<- NA
  taxa_table[taxa_table=="Unidentified"]<- NA
  taxa_table[taxa_table==""]<- NA
  taxa_table[taxa_table=="Incertae sedis"]<- NA
  taxa_table[taxa_table=="uncultured bacterium"]<- NA
  #to remove species that aready have sp
  taxa_table$Species <- 
    gsub(" sp ", "", taxa_table$Species)
  # add sp. to species with just the Genus
  taxa_table[which(is.na(taxa_table$Genus) == FALSE),]$Genus <-
    paste(taxa_table$Genus[is.na(taxa_table$Genus) == FALSE], "sp.", sep = " ")
    taxa_table <- taxa_table[c(8,1,2,3,4,5,6,7)]
  taxa_table[] = lapply(taxa_table, blank2na, na.strings=c('','NA','na','N/A','n/a','NaN','nan'))
  lastValue <- function(x) tail(x[!is.na(x)], 1)
  last_taxons<- apply(taxa_table[,1:8], 1, lastValue)
  taxa_table$BestMatch <- last_taxons
  taxa_table[, "BestMatch"] <-
    gsub("_", " ", taxa_table[, "BestMatch"])
  taxa_table$Label <-
    paste(taxa_table$OTU, taxa_table$BestMatch, sep = "_")
  taxa_table[, "Genus"] <- gsub(" sp.", "", taxa_table[, "Genus"])
  tax_table(dataframe) <- tax_table(as.matrix(taxa_table))
  return(dataframe)
}
```
Only run on file once!
```{r}
#ReformatTaxonomy(fun.pruned) -> fun_psq

# Save CSS Normalized, Filtered, and Renamed RDS file #
#saveRDS(fun_psq, file = "Fungi_RDS/Spermosphere_Fungi_CSS_filt.rds")

# Restore the object
#fun_filt <- readRDS(file = "Fungi_RDS/Spermosphere_fungi_CSS_filt.rds")
```


Top taxa in the Spermosphere samples
```{r}
fun.css.norm2 <- fun_psq %>%
  subset_samples(Sample_Type == "Spermosphere") %>%
  transform_sample_counts(function(x) {x/sum(x)} )

topx.fungi <- top_taxa(fun.css.norm2, n = 40)

fun.composition <- fun.css.norm2 %>%
  subset_taxa(OTU_ID %in% topx.fungi) %>%
  microbiome::transform("compositional") %>%
  psmelt() %>% 
  group_by(Location_EC, Time, Label) %>%
  mutate(Time = factor(Time, levels = c("0", "8", "12", "16"))) %>%
  summarise(MeanRelAbund = mean(Abundance)) %>%
  arrange(-MeanRelAbund) %>%
  left_join(as.data.frame(tax_table(fun.css.norm2), by = "Label")) %>% 
  ggplot(aes(Location_EC, MeanRelAbund, fill = Label)) +
  geom_bar(stat = "identity") +
  theme_classic() +
  scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb, col_vector)) +
  scale_y_continuous(labels = scales::percent) +
  labs(x = "", y = "Relative abundance (%)",
       title = "Top 40 taxa") + 
  theme(axis.text = element_text(size = 12, angle = 45, vjust = 1, hjust = 1),
        axis.title = element_text(size = 12), 
        strip.text = element_text(size = 12), plot.title = element_text(size = 12),
        legend.text = element_text(face = "italic", size = 12), 
        legend.title = element_blank()#,plot.margin = margin(t = 60)
        ) +
  facet_wrap(~Time)
fun.composition

#ggsave(filename = "bar2_usearch25.png", plot = fun.composition, device = "png", width = 250, height = 200, units="mm", dpi = 600)
```

```