---
title: "Bacterial_Diversity_Analysis"
author: "Logan Luchs"
date: "2025-02-22"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
setwd("Bacteria/3_Alpha_and_Beta_Diversity")
```

#Data Preprocessing#

## Libraries ##
```{r}
library(BiocManager)
library(phyloseq)
library(vegan)
library(tidyverse)
library(metagenomeSeq)
library(ggpubr)
library(Biostrings)
library(microbiome)
library(ggrepel)
library(decontam)
library(picante) #For Faiths Phylogentic Diversity
library(MASS) #For Linear Modeling
library(betareg) #For Linear Modeling
library(emmeans)
```

## Set global options ##
```{r}
# Set global options #
# no scientific notation
options(scipen=10000) 

# color blind pallets used throughout 
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ibm.cbb <- c("#648FFF", "#785EF0", "#DC267F", "#FE6100", "#FFB000")
tol.cbb <- c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255")
```

##NOTE: Rmd files set the working directory to the place where the rmd file is, not the actual working directory
##Read in and Separation##
## Read in the RDS files #
```{r}
###Bacterial RDS
BSpermosphere_RDS_CSS <- readRDS("Bacteria_RDS/Spermosphere_CSS.rds") ##Using Normalized Reads

BSpermosphere_RDS_NoNorm <- readRDS("Bacteria_RDS/Spermosphere_bac_clean.rds") ##Using Non-normalized Reads
```


##Seperation of Nonsteamed vs Steamed soils##
```{r}
bac.Auto <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "Autoclaved_Potting_Soil")
bac.EV <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "EVSmith_EC")
bac.MI <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "Michigan")
bac.ND <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "N_Dakota")
bac.Wire <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "Wiregrass_EC")
bac.PA <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "Pennsylvania")
bac.AR <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "Arkansas")
bac.TVR <- subset_samples(BSpermosphere_RDS_NoNorm, Location_EC == "TVRec_EC")

view(bac.EV@sam_data)
```

## separating the otu table, metadata, and the tax data for ease #
```{r}
#BACTERIA Steamed
Sbac.meta.data <- data.frame(bac.Steamed@sam_data)
Sbac.tax.data <- data.frame(bac.Steamed@tax_table)
Sbac.otu <- bac.Steamed@otu_table %>%
  as("matrix")
#BACTERIA NonSteamed
Nbac.meta.data <- data.frame(bac.Nonsteamed@sam_data)
Nbac.tax.data <- data.frame(bac.Nonsteamed@tax_table)
Nbac.otu <- bac.Nonsteamed@otu_table %>%
  as("matrix")

cssbac.meta.data <- data.frame(BSpermosphere_RDS_CSS@sam_data)



```


# Alpha Diversity Analysis #
## Autoclaved Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.Auto@sam_data$shannon <- estimate_richness(bac.Auto, measures=c("Shannon"))$Shannon
bac.Auto@sam_data$invsimpson <- estimate_richness(bac.Auto, measures=c("InvSimpson"))$InvSimpson
bac.Auto@sam_data$simpson <- estimate_richness(bac.Auto, measures=c("Simpson"))$Simpson
bac.Auto@sam_data$richness <- estimate_richness(bac.Auto, measures=c("Observed"))$Observed
bac.Auto@sam_data$even <- bac.Auto@sam_data$shannon/log(bac.Auto@sam_data$richness)
Autosample.data.bac <- data.frame(bac.Auto@sam_data)
Autosample.data.bac$Time <- factor(Autosample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
Auto_otu <- bac.Auto@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
autophylo.diversity <- pd(t(Auto_otu), bac.Auto@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
autophylo.diversity$Samples <- rownames(autophylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
autophylo.diversity.join <- left_join(autophylo.diversity, as.data.frame(bac.Auto@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
autophylo.diversity.join$Time.Point <- factor(autophylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```
## EVSMITH Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.EV@sam_data$shannon <- estimate_richness(bac.EV, measures=c("Shannon"))$Shannon
bac.EV@sam_data$invsimpson <- estimate_richness(bac.EV, measures=c("InvSimpson"))$InvSimpson
bac.EV@sam_data$simpson <- estimate_richness(bac.EV, measures=c("Simpson"))$Simpson
bac.EV@sam_data$richness <- estimate_richness(bac.EV, measures=c("Observed"))$Observed
bac.EV@sam_data$even <- bac.EV@sam_data$shannon/log(bac.EV@sam_data$richness)
EVsample.data.bac <- data.frame(bac.EV@sam_data)
EVsample.data.bac$Time <- factor(EVsample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
EV_otu <- bac.EV@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
EVphylo.diversity <- pd(t(EV_otu), bac.EV@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
EVphylo.diversity$Samples <- rownames(EVphylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
EVphylo.diversity.join <- left_join(EVphylo.diversity, as.data.frame(bac.EV@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
EVphylo.diversity.join$Time.Point <- factor(EVphylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```
## Michigan Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.MI@sam_data$shannon <- estimate_richness(bac.MI, measures=c("Shannon"))$Shannon
bac.MI@sam_data$invsimpson <- estimate_richness(bac.MI, measures=c("InvSimpson"))$InvSimpson
bac.MI@sam_data$simpson <- estimate_richness(bac.MI, measures=c("Simpson"))$Simpson
bac.MI@sam_data$richness <- estimate_richness(bac.MI, measures=c("Observed"))$Observed
bac.MI@sam_data$even <- bac.MI@sam_data$shannon/log(bac.MI@sam_data$richness)
MIsample.data.bac <- data.frame(bac.MI@sam_data)
MIsample.data.bac$Time <- factor(MIsample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
MI_otu <- bac.MI@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
MIphylo.diversity <- pd(t(MI_otu), bac.MI@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
MIphylo.diversity$Samples <- rownames(MIphylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
MIphylo.diversity.join <- left_join(MIphylo.diversity, as.data.frame(bac.MI@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
MIphylo.diversity.join$Time.Point <- factor(MIphylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```
## North Dakota Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.ND@sam_data$shannon <- estimate_richness(bac.ND, measures=c("Shannon"))$Shannon
bac.ND@sam_data$invsimpson <- estimate_richness(bac.ND, measures=c("InvSimpson"))$InvSimpson
bac.ND@sam_data$simpson <- estimate_richness(bac.ND, measures=c("Simpson"))$Simpson
bac.ND@sam_data$richness <- estimate_richness(bac.ND, measures=c("Observed"))$Observed
bac.ND@sam_data$even <- bac.ND@sam_data$shannon/log(bac.ND@sam_data$richness)
NDsample.data.bac <- data.frame(bac.ND@sam_data)
NDsample.data.bac$Time <- factor(NDsample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
ND_otu <- bac.ND@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
NDphylo.diversity <- pd(t(ND_otu), bac.ND@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
NDphylo.diversity$Samples <- rownames(NDphylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
NDphylo.diversity.join <- left_join(NDphylo.diversity, as.data.frame(bac.ND@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
NDphylo.diversity.join$Time.Point <- factor(NDphylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```
## Wiregrass Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.Wire@sam_data$shannon <- estimate_richness(bac.Wire, measures=c("Shannon"))$Shannon
bac.Wire@sam_data$invsimpson <- estimate_richness(bac.Wire, measures=c("InvSimpson"))$InvSimpson
bac.Wire@sam_data$simpson <- estimate_richness(bac.Wire, measures=c("Simpson"))$Simpson
bac.Wire@sam_data$richness <- estimate_richness(bac.Wire, measures=c("Observed"))$Observed
bac.Wire@sam_data$even <- bac.Wire@sam_data$shannon/log(bac.Wire@sam_data$richness)
Wiresample.data.bac <- data.frame(bac.Wire@sam_data)
Wiresample.data.bac$Time <- factor(Wiresample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
Wire_otu <- bac.Wire@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
Wirephylo.diversity <- pd(t(Wire_otu), bac.Wire@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
Wirephylo.diversity$Samples <- rownames(Wirephylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
Wirephylo.diversity.join <- left_join(Wirephylo.diversity, as.data.frame(bac.Wire@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
Wirephylo.diversity.join$Time.Point <- factor(Wirephylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```
## Pennsylvania Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.PA@sam_data$shannon <- estimate_richness(bac.PA, measures=c("Shannon"))$Shannon
bac.PA@sam_data$invsimpson <- estimate_richness(bac.PA, measures=c("InvSimpson"))$InvSimpson
bac.PA@sam_data$simpson <- estimate_richness(bac.PA, measures=c("Simpson"))$Simpson
bac.PA@sam_data$richness <- estimate_richness(bac.PA, measures=c("Observed"))$Observed
bac.PA@sam_data$even <- bac.PA@sam_data$shannon/log(bac.PA@sam_data$richness)
PAsample.data.bac <- data.frame(bac.PA@sam_data)
PAsample.data.bac$Time <- factor(PAsample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
PA_otu <- bac.PA@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
PAphylo.diversity <- pd(t(PA_otu), bac.PA@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
PAphylo.diversity$Samples <- rownames(PAphylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
PAphylo.diversity.join <- left_join(PAphylo.diversity, as.data.frame(bac.PA@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
PAphylo.diversity.join$Time.Point <- factor(PAphylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```
## Arkansas Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.AR@sam_data$shannon <- estimate_richness(bac.AR, measures=c("Shannon"))$Shannon
bac.AR@sam_data$invsimpson <- estimate_richness(bac.AR, measures=c("InvSimpson"))$InvSimpson
bac.AR@sam_data$simpson <- estimate_richness(bac.AR, measures=c("Simpson"))$Simpson
bac.AR@sam_data$richness <- estimate_richness(bac.AR, measures=c("Observed"))$Observed
bac.AR@sam_data$even <- bac.AR@sam_data$shannon/log(bac.AR@sam_data$richness)
ARsample.data.bac <- data.frame(bac.AR@sam_data)
ARsample.data.bac$Time <- factor(ARsample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
AR_otu <- bac.AR@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
ARphylo.diversity <- pd(t(AR_otu), bac.AR@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
ARphylo.diversity$Samples <- rownames(ARphylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
ARphylo.diversity.join <- left_join(ARphylo.diversity, as.data.frame(bac.AR@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
ARphylo.diversity.join$Time.Point <- factor(ARphylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```
## TVREC Soil ##
```{r}
### Calculating Shannon, Simpson, Inverse Simpson and Pielou's Eveness ###

bac.TVR@sam_data$shannon <- estimate_richness(bac.TVR, measures=c("Shannon"))$Shannon
bac.TVR@sam_data$invsimpson <- estimate_richness(bac.TVR, measures=c("InvSimpson"))$InvSimpson
bac.TVR@sam_data$simpson <- estimate_richness(bac.TVR, measures=c("Simpson"))$Simpson
bac.TVR@sam_data$richness <- estimate_richness(bac.TVR, measures=c("Observed"))$Observed
bac.TVR@sam_data$even <- bac.TVR@sam_data$shannon/log(bac.TVR@sam_data$richness)
TVRsample.data.bac <- data.frame(bac.TVR@sam_data)
TVRsample.data.bac$Time <- factor(TVRsample.data.bac$Time, levels = c("0", "8", "12","16"))
### Calculating Faiths Phylogenetic Diversity ###
TVR_otu <- bac.TVR@otu_table %>%
  as.data.frame()

# Calculate Faith's phylogenetic diversity for each sample, excluding the root
TVRphylo.diversity <- pd(t(TVR_otu), bac.TVR@phy_tree, include.root = F)

# Add sample names as a column in the phylogenetic diversity data frame
TVRphylo.diversity$Samples <- rownames(TVRphylo.diversity)

# Join the phylogenetic diversity data frame with the sample data from 'bac.Nonsteamed' based on sample names
TVRphylo.diversity.join <- left_join(TVRphylo.diversity, as.data.frame(bac.TVR@sam_data), by = c("Samples" = "Sample_ID"))

# Convert the 'Time' column in the joined data frame to a factor with specified levels
TVRphylo.diversity.join$Time.Point <- factor(TVRphylo.diversity.join$Time, levels = c("0", "8", "12","16"))

```


###Merging Steamed and Nonsteamed Alpha Diversity Data Frames###
```{r}
autophylo.diversity.join$Location2 <- "Autoclaved_Control"
MIphylo.diversity.join$Location2 <- "Michigan"
NDphylo.diversity.join$Location2 <- "North_Dakota"
Wirephylo.diversity.join$Location2 <- "Wiregrass"
PAphylo.diversity.join$Location2 <- "Pennsylvania"
ARphylo.diversity.join$Location2 <- "Arkansas"
EVphylo.diversity.join$Location2 <- "EV_Smith"
TVRphylo.diversity.join$Location2 <- "TVREC"
# Combine the 'Sbphylo.diversity.join' and 'nbphylo.diversity.join' data frames into one data frame 'BAlphaCombined'
BAlphaCombined <- rbind.data.frame(autophylo.diversity.join, MIphylo.diversity.join,NDphylo.diversity.join, Wirephylo.diversity.join, PAphylo.diversity.join, ARphylo.diversity.join, EVphylo.diversity.join, TVRphylo.diversity.join)
```

###Plotting Richness for Prokaryotes as a Box-Plot###
```{r}
bac.richness <- BAlphaCombined %>%
  ggplot(aes(x = Sample_Type, y = richness, color = Time)) +
  facet_wrap(~factor(Location2)) +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  geom_pwc(aes(group = Location2), method = "t_test", label = "p.adj.format") +
  #stat_compare_means(method = "t.test") +
  scale_y_continuous(limits = c(0, 5000), breaks = seq(0, 5000, 500)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun = mean,geom="line") +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Richness") +
  xlab("Type of Environment")+
  #geom_jitter(width = 0.1, alpha = 0.8)+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  #stat_compare_means(method = "kruskal", hide.ns = TRUE) +
  theme_classic()+
  theme(legend.text = element_text(face = "italic", size = 12), legend.title = element_blank())
bac.richness

bac.richness <- BAlphaCombined %>%
  ggplot(aes(x = Sample_Type, y = richness, color = Location2)) +
  facet_wrap(~factor(Time, levels = c("0", "8", "12","16"))) +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  geom_pwc(aes(group = Location2), method = "t_test", label = "p.adj.format") +
  #stat_compare_means(method = "t.test") +
  scale_y_continuous(limits = c(0, 5000), breaks = seq(0, 5000, 500)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun = mean,geom="line") +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Richness") +
  xlab("Type of Environment")+
  #geom_jitter(width = 0.1, alpha = 0.8)+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  #stat_compare_means(method = "kruskal", hide.ns = TRUE) +
  theme_classic()+
  theme(legend.text = element_text(face = "italic", size = 12), legend.title = element_blank())
bac.richness


#This code creates a ggplot object to visualize the richness data from the BAlphaCombined data frame. It sets the x-axis to represent the 'Type' of environment and the y-axis to show the 'richness' values, with colors distinguishing between "Steamed" and "NonSteamed" samples. The plot is divided into separate panels for each time point ("Planting," "17hrs," and "V2") using facet_wrap. Boxplots are added to display the distribution of richness values, with individual data points overlaid using jitter to avoid overlap. The y-axis is scaled to range from 0 to 4000, with breaks at every 500 units. A t-test is performed to compare means, and the results are displayed on the plot. The plot is customized with specific labels for the x and y axes, and a white background theme is applied. Additionally, the legend text is italicized, and the legend title is removed for a cleaner appearance. The final plot is stored in the bac.richness object and displayed.


```

###Plotting Faith's Phylogenetic Diversity for Prokaryotes as a Box-Plot###
```{r}
bac.phylo.div <- BAlphaCombined %>%
  ggplot(aes(x = Sample_Type, y = PD, color = Location2)) +
  facet_wrap(~factor(Time, levels = c("0", "8", "12","16")), scale = "free") +
  #stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
    #scale_y_continuous(limits = c(30, 210), breaks = seq(30, 210, 70)) +
  geom_point(position=position_jitterdodge(0.05)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun = mean,geom="line") +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Faith's PD") +
  xlab("Type of Environment")+
  #geom_jitter(width = 0.1, alpha = 0.8)+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  #stat_compare_means(method = "kruskal", hide.ns = TRUE) +
  theme_classic()+
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
bac.phylo.div

bac.phylo.div <- BAlphaCombined %>%
  ggplot(aes(x = Sample_Type, y = PD, color = Time)) +
  facet_wrap(~factor(Location2), scale = "free") +
  #stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
    #scale_y_continuous(limits = c(30, 210), breaks = seq(30, 210, 70)) +
  geom_point(position=position_jitterdodge(0.05)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun = mean,geom="line") +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Faith's PD") +
  xlab("Type of Environment")+
  #geom_jitter(width = 0.1, alpha = 0.8)+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  #stat_compare_means(method = "kruskal", hide.ns = TRUE) +
  theme_classic()+
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
bac.phylo.div
#This code creates a ggplot object to visualize Faith's phylogenetic diversity (PD) from the BAlphaCombined data frame. It sets the x-axis to represent the 'Type' of environment and the y-axis to show the PD values, with colors distinguishing between "Steamed" and "NonSteamed" samples. The plot is divided into separate panels for each time point ("Planting," "17hrs," and "V2") using facet_wrap. Boxplots are added to display the distribution of PD values, with individual data points overlaid using jitter to avoid overlap. The y-axis is scaled to range from 30 to 210, with breaks at every 70 units. The plot is customized with specific labels for the x and y axes, and a classic theme is applied. Additionally, the legend text is italicized, and the legend title is removed for a cleaner appearance. The final plot is stored in the bac.phylo.div object and displayed.
```

###Plotting Pielou's Eveness for Prokaryotes as a Box-Plot###
```{r}
bac.even <- BAlphaCombined %>%
  ggplot(aes(x = Sample_Type, y = even, color = Location2)) +
  facet_wrap(~factor(Time, levels = c("0", "8", "12","16")), scale = "free") +
  #scale_y_continuous(limits = c(0.45, 0.85), breaks = seq(0.45, 0.85, 0.05)) +
  #stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun.y=mean,geom="bar", position = "dodge", width = 0.5) +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Pielou's evenness") +
  xlab("Type of Environment")+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  theme_classic() +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
bac.even
bac.even <- BAlphaCombined %>%
  ggplot(aes(x = Sample_Type, y = even, color = Time)) +
  facet_wrap(~factor(Location2)) +
  #scale_y_continuous(limits = c(0.45, 0.85), breaks = seq(0.45, 0.85, 0.05)) +
  #stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun.y=mean,geom="bar", position = "dodge", width = 0.5) +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Pielou's evenness") +
  xlab("Type of Environment")+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  theme_classic() +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
bac.even
#This code creates a ggplot object to visualize Pielou's evenness from the BAlphaCombined data frame. It sets the x-axis to represent the 'Type' of environment and the y-axis to show the evenness values, with colors distinguishing between "Steamed" and "NonSteamed" samples. The plot is divided into separate panels for each time point ("Planting," "17hrs," and "V2") using facet_wrap. Boxplots are added to display the distribution of evenness values, with individual data points overlaid using jitter to avoid overlap. The y-axis is scaled to range from 0.45 to 0.85, with breaks at every 0.05 units. The plot is customized with specific labels for the x and y axes, and a classic theme is applied. Additionally, the legend text is italicized, and the legend title is removed for a cleaner appearance. The final plot is stored in the bac.even object and displayed.

```
#### Richness statistics ####
```{r}

# for every X - unit change in X we observed a exp^b times as many species.  
# 1 exp^b
# 2. Times as many for counts 
# 3. report confidence interval - exp^lower to exp^upper

### calculating observed means ###

ObservedRichness <- BAlphaCombined %>%
  group_by(Trt, Time, Type.revised) %>%  
  summarise(MeanRichness = mean(richness), STDRichness = sd(richness)) %>% 
  arrange(-MeanRichness)

### Richness Statistics ###

# Since the data is may not normally distributed because and it is count data. Fit a Poisson regression model to predict 'richness' based on 'Type.revised', 'Time', 'Trt', and their interactions
RichPoisson <- glm(richness ~ Sample_Type + Time + Location2 + Location2 * Time * Sample_Type, data = BAlphaCombined, family = "poisson")

# Perform an analysis of variance (ANOVA) on the fitted model
anova(RichPoisson)

# Display a summary of the fitted model, including coefficients, standard errors, z-values, and p-values
summary(RichPoisson)

# Calculate the variance by dividing the deviance by the residual degrees of freedom
RichPoisson$deviance / RichPoisson$df.residual
#output is 38.20, way higher than what is recommended so negative binomial may be better, the target is to get close to 1.

# To fix the distribution variance, Fit a negative binomial regression model to predict 'richness' based on 'Type.revised', 'Time', 'Trt', and their interactions
RichStat <- glm.nb(richness ~ Sample_Type + Time + Location2 + Location2 * Time * Sample_Type, data = BAlphaCombined)
# Perform an analysis of variance (ANOVA) on the fitted model
anova(RichStat)

# Display a summary of the fitted model, including coefficients, standard errors, z-values, and p-values
summary(RichStat)

# Calculate the variance by dividing the deviance by the residual degrees of freedom
RichStat$deviance / RichStat$df.residual
 
# Mean separation
# Calculate estimated marginal means (least-squares means) for the 'RichStat' model, examining the effect of treatment within combinations of type and time
lsmeans <- emmeans(RichStat, ~Location2 | Sample_Type * Time)

# Perform multiple comparisons with Tukey adjustment and generate compact letter display (CLD) for the estimated marginal means
library(multcompView)
Results_lsmeansEC <- multcomp::cld(lsmeans, alpha = 0.05, reversed = TRUE, details = TRUE, Letters = letters)

```

#### Evenness Statstics ####
```{r}
##Evenness Statistics
#install.packages("betareg")
library(betareg)
# Fit a beta regression model to eveness using specified predictors
EvenStat.B <- betareg(even ~ Sample_Type + Time + Location2 + Location2 * Time * Sample_Type, data = BAlphaCombined)
#works because the data ranges 0-1

# Perform an ANOVA on the fitted model
anova(EvenStat.B)

# Display a summary of the fitted model
summary(EvenStat.B)

# Mean separation
# Calculate estimated marginal means (least-squares means) for the treatment effect within combinations of type and time
lsmeans <- emmeans(EvenStat.B, ~Location2 | Sample_Type * Time)

# Perform multiple comparisons with Tukey adjustment
Results_lsmeansEC <- multcomp::cld(lsmeans, alpha = 0.05, reversed = TRUE, details = TRUE, Letters = letters)

# Calculate observed evenness statistics
ObservedEveness <- BAlphaCombined %>%
  group_by(Location2, Time, Sample_Type) %>%   
  summarise(MeanEvenness = mean(even), STDEvenness = sd(even)) %>% 
  arrange(-MeanEvenness)
```

## Faith's Phylogenetic Diversity Statistics
```{r}
# Fit a negative binomial regression model to the 'PD' (Faiths Phylogenetic Diversity) variable using the specified predictors
RichStat.C <- glm.nb(PD ~ Sample_Type + Time + Location2 + Location2 * Time * Sample_Type, data = BAlphaCombined) 

# Perform an ANOVA on the fitted model
anova(RichStat.C)

# Display a summary of the fitted model
summary(RichStat.C)

# Calculate the ratio of deviance to degrees of freedom for the fitted model
RichStat.C$deviance / RichStat.C$df.residual

# Mean separation
# Calculate estimated marginal means (least-squares means) for the treatment effect within combinations of type and time
lsmeans <- emmeans(RichStat.C, ~Location2 | Sample_Type * Time)

# Perform multiple comparisons with Tukey adjustment and display results
Results_lsmeansEC <- multcomp::cld(lsmeans, alpha = 0.05, reversed = TRUE, details = TRUE, Letters = letters)

# Calculate observed Faith's Phylogenetic Diversity statistics
ObservedFPD <- BAlphaCombined %>%
  group_by(Location2, Time, Sample_Type) %>%   
  summarise(MeanFPD = mean(PD), STFPD = sd(PD)) %>% 
  arrange(-MeanFPD)

```
####Conjoined Figure
```{r}
figure1b <- ggpubr::ggarrange(bac.richness,
                                       bac.even,
                                       bac.phylo.div,
                                       labels = "auto",
                                       nrow = 3, ncol = 1, common.legend = T)

#Arranging the figures

```

# Beta Diversity
## Bacterial Beta-diversity
###### Beta-Diversity using Weighted Unifrac Only Bacteria ####
```{r}
##### All Samples ########
bac.unifrac = UniFrac(BSpermosphere_RDS_CSS, weighted = TRUE) # create weighted unifrac distance matrix

# PERMANOVA - testing for differences in centroids
set.seed(12345)  # Set seed for reproducibility
unifracstats <- adonis2(bac.unifrac ~ Location_EC * Sample_Type * Time, as(sample_data(BSpermosphere_RDS_CSS), "data.frame"))  # Perform PERMANOVA
#the model is significant therefore we will subset by time.

# Perform ordination using MDS with weighted UniFrac distance
global.ord.uni <- phyloseq::ordinate(BSpermosphere_RDS_CSS, "MDS", "unifrac", weighted = TRUE)

# Calculate percentage variation on axis one
variation.axis1 <- round(100 * global.ord.uni$values$Relative_eig[[1]], 2)

# Calculate percentage variation on axis two
variation.axis2 <- round(100 * global.ord.uni$values$Relative_eig[[2]], 2)

# Extract ordination vectors
bac.ggplot.data.uni <- data.frame(global.ord.uni$vectors)
bac.ggplot.data.uni$Sample_ID <- rownames(bac.ggplot.data.uni)

# Merge ordination vectors with sample metadata
bac.ggplot.data.uni2 <- left_join(data.frame(BSpermosphere_RDS_CSS@sam_data), bac.ggplot.data.uni, by = "Sample_ID")

# Order the time points in chronological order
bac.ggplot.data.uni2$Time.Point <- factor(bac.ggplot.data.uni2$Time, levels = c("0", "8", "12", "16"))

# Create a ggplot for weighted UniFrac distance

global.uni <- ggplot(bac.ggplot.data.uni2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type, color = Location_EC)) +
  geom_point(size = 4, alpha = 0.7) +  # Add points with specified size and transparency
  scale_shape_manual(values = c(21, 22, 24, 25)) +  # Customize shapes for different types
  scale_fill_manual(values = cbbPalette) +  # Customize fill colors for different treatments
  scale_color_manual(values = cbbPalette) +  # Customize colors for different treatments
  xlab(paste("PcoA1 -", variation.axis1, "%")) +  # Label for x-axis with percentage variation 1
  ylab(paste("PcoA2 -", variation.axis2, "%")) +  # Label for y-axis with percentage variation 2
  theme(legend.position = "bottom", plot.title = element_text(size = 10, face = "bold")) +  # Customize theme
  ggtitle("Weighted Unifrac")  # Add title to the plot
global.uni  # Display the plot


```
### 0 hours; Unifrac #####
```{r}
# Filter to time point
bac_sperm_0 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "0") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)  # Set seed for reproducibility
prok.dist.uni.0hrs = UniFrac(bac_sperm_0, weighted = TRUE)  # Create distance matrix
adonis.0.uni <- adonis2(prok.dist.uni.0hrs ~ Location_EC * Sample_Type, as(sample_data(bac_sperm_0), "data.frame"))  # Perform PERMANOVA to test for significant changes
adonis.0.uni  # Display PERMANOVA results

# The R2 is the variation due to different factors 
# pvalue is the p-value for the factor
pvalue <- adonis.0.uni$`Pr(>F)`[[1]]
R2 <- adonis.0.uni$R2[[1]]

# Plotting
ordination.unifrac.pcoa.0hrs <- ordinate(bac_sperm_0, "PCoA", distance = prok.dist.uni.0hrs)  # Calculate the resemblance and ordinate using PCoA
uni.0.values <- data.frame(ordination.unifrac.pcoa.0hrs$values)

# Percent variation
variation.axis1.0.uni <- round(100 * uni.0.values$Relative_eig[[1]], 2)
variation.axis2.0.uni <- round(100 * uni.0.values$Relative_eig[[2]], 2)

# Data from ordinations
uni.0.vectors <- data.frame(ordination.unifrac.pcoa.0hrs$vectors)  # Positions of your points on the PCoA graph
uni.0.vectors$Sample_ID <- rownames(uni.0.vectors)
uni.0.vectors2 <- left_join(data.frame(bac_sperm_0@sam_data), uni.0.vectors, by = "Sample_ID")

# Create a ggplot for UniFrac distance at 0 hours
pcoa.unifrac.0hrs.new <- ggplot(uni.0.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size = 4, alpha = 0.7) +
  scale_shape_manual(values = c(21, 24)) +
  scale_fill_manual(values = cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.0.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.0.uni, "%")) +
  guides(fill = guide_legend(override.aes = list(shape = 21))) +
  ggtitle(paste("Planting")) +
  theme_bw() +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.key = element_blank(),
        plot.title = element_text(size = 10, face = "bold"))
pcoa.unifrac.0hrs.new

# Dispersion for each Time-Point
beta.disp.0.uni <- betadisper(prok.dist.uni.0hrs, bac_sperm_0@sam_data$Location_EC)
permutest(beta.disp.0.uni)
beta.disp.0.unib <- betadisper(prok.dist.uni.0hrs, bac_sperm_0@sam_data$Sample_Type)
permutest(beta.disp.0.unib)
```
### 8 hours; Unifrac #####
```{r}
# Filter to time point
bac_sperm_8 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "8") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)  # Set seed for reproducibility
prok.dist.uni.8hrs = UniFrac(bac_sperm_8, weighted = TRUE)  # Create distance matrix
adonis.8.uni <- adonis2(prok.dist.uni.8hrs ~ Location_EC * Sample_Type, as(sample_data(bac_sperm_8), "data.frame"))  # Perform PERMANOVA to test for significant changes
adonis.8.uni  # Display PERMANOVA results

# The R2 is the variation due to different factors 
# pvalue is the p-value for the factor
pvalue <- adonis.8.uni$`Pr(>F)`[[1]]
R2 <- adonis.8.uni$R2[[1]]

# Plotting
ordination.unifrac.pcoa.8hrs <- ordinate(bac_sperm_8, "PCoA", distance = prok.dist.uni.8hrs)  # Calculate the resemblance and ordinate using PCoA
uni.8.values <- data.frame(ordination.unifrac.pcoa.8hrs$values)

# Percent variation
variation.axis1.8.uni <- round(100 * uni.8.values$Relative_eig[[1]], 2)
variation.axis2.8.uni <- round(100 * uni.8.values$Relative_eig[[2]], 2)

# Data from ordinations
uni.8.vectors <- data.frame(ordination.unifrac.pcoa.8hrs$vectors)  # Positions of your points on the PCoA graph
uni.8.vectors$Sample_ID <- rownames(uni.8.vectors)
uni.8.vectors2 <- left_join(data.frame(bac_sperm_8@sam_data), uni.8.vectors, by = "Sample_ID")

# Create a ggplot for UniFrac distance at 8 hours
pcoa.unifrac.8hrs.new <- ggplot(uni.8.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size = 4, alpha = 0.7) +
  scale_shape_manual(values = c(21, 24)) +
  scale_fill_manual(values = cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.8.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.8.uni, "%")) +
  guides(fill = guide_legend(override.aes = list(shape = 21))) +
  ggtitle(paste("8 Hours Postplanting")) +
  theme_bw() +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.key = element_blank(),
        plot.title = element_text(size = 10, face = "bold"))
pcoa.unifrac.8hrs.new

# Dispersion for each Time-Point
beta.disp.8.uni <- betadisper(prok.dist.uni.8hrs, bac_sperm_8@sam_data$Location_EC)
permutest(beta.disp.8.uni)
beta.disp.8.unib <- betadisper(prok.dist.uni.8hrs, bac_sperm_8@sam_data$Sample_Type)
permutest(beta.disp.8.unib)
```
### 12; Unifrac #####
```{r}
# Filter to time point
bac_sperm_12 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "12") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)  # Set seed for reproducibility
prok.dist.uni.12hrs = UniFrac(bac_sperm_12, weighted = TRUE)  # Create distance matrix
adonis.12.uni <- adonis2(prok.dist.uni.12hrs ~ Location_EC * Sample_Type, as(sample_data(bac_sperm_12), "data.frame"))  # Perform PERMANOVA to test for significant changes
adonis.12.uni  # Display PERMANOVA results

# The R2 is the variation due to different factors 
# pvalue is the p-value for the factor
pvalue <- adonis.12.uni$`Pr(>F)`[[1]]
R2 <- adonis.12.uni$R2[[1]]

# Plotting
ordination.unifrac.pcoa.12hrs <- ordinate(bac_sperm_12, "PCoA", distance = prok.dist.uni.12hrs)  # Calculate the resemblance and ordinate using PCoA
uni.12.values <- data.frame(ordination.unifrac.pcoa.12hrs$values)

# Percent variation
variation.axis1.12.uni <- round(100 * uni.12.values$Relative_eig[[1]], 2)
variation.axis2.12.uni <- round(100 * uni.12.values$Relative_eig[[2]], 2)

# Data from ordinations
uni.12.vectors <- data.frame(ordination.unifrac.pcoa.12hrs$vectors)  # Positions of your points on the PCoA graph
uni.12.vectors$Sample_ID <- rownames(uni.12.vectors)
uni.12.vectors2 <- left_join(data.frame(bac_sperm_12@sam_data), uni.12.vectors, by = "Sample_ID")

# Create a ggplot for UniFrac distance at 12 growth stage
pcoa.unifrac.12hrs.new <- ggplot(uni.12.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size = 4, alpha = 0.7) +
  scale_shape_manual(values = c(21, 24)) +
  scale_fill_manual(values = cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.12.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.12.uni, "%")) +
  guides(fill = guide_legend(override.aes = list(shape = 21))) +
  ggtitle(paste("12 Growth Stage")) +
  theme_bw() +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.key = element_blank(),
        plot.title = element_text(size = 10, face = "bold"))
pcoa.unifrac.12hrs.new

# Dispersion for each Time-Point
beta.disp.12.uni <- betadisper(prok.dist.uni.12hrs, bac_sperm_12@sam_data$Location_EC)
permutest(beta.disp.12.uni)
beta.disp.12.unib <- betadisper(prok.dist.uni.12hrs, bac_sperm_12@sam_data$Sample_Type)
permutest(beta.disp.12.unib)

```
### 16; Unifrac #####
```{r}
# Filter to time point
bac_sperm_16 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "16") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)  # Set seed for reproducibility
prok.dist.uni.16hrs = UniFrac(bac_sperm_16, weighted = TRUE)  # Create distance matrix
adonis.16.uni <- adonis2(prok.dist.uni.16hrs ~ Location_EC * Sample_Type, as(sample_data(bac_sperm_16), "data.frame"))  # Perform PERMANOVA to test for significant changes
adonis.16.uni  # Display PERMANOVA results

# The R2 is the variation due to different factors 
# pvalue is the p-value for the factor
pvalue <- adonis.16.uni$`Pr(>F)`[[1]]
R2 <- adonis.16.uni$R2[[1]]

# Plotting
ordination.unifrac.pcoa.16hrs <- ordinate(bac_sperm_16, "PCoA", distance = prok.dist.uni.16hrs)  # Calculate the resemblance and ordinate using PCoA
uni.16.values <- data.frame(ordination.unifrac.pcoa.16hrs$values)

# Percent variation
variation.axis1.16.uni <- round(100 * uni.16.values$Relative_eig[[1]], 2)
variation.axis2.16.uni <- round(100 * uni.16.values$Relative_eig[[2]], 2)

# Data from ordinations
uni.16.vectors <- data.frame(ordination.unifrac.pcoa.16hrs$vectors)  # Positions of your points on the PCoA graph
uni.16.vectors$Sample_ID <- rownames(uni.16.vectors)
uni.16.vectors2 <- left_join(data.frame(bac_sperm_16@sam_data), uni.16.vectors, by = "Sample_ID")

# Create a ggplot for UniFrac distance at 16 growth stage
pcoa.unifrac.16hrs.new <- ggplot(uni.16.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size = 4, alpha = 0.7) +
  scale_shape_manual(values = c(21, 24)) +
  scale_fill_manual(values = cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.16.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.16.uni, "%")) +
  guides(fill = guide_legend(override.aes = list(shape = 21))) +
  ggtitle(paste("16 Growth Stage")) +
  theme_bw() +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.key = element_blank(),
        plot.title = element_text(size = 10, face = "bold"))
pcoa.unifrac.16hrs.new

# Dispersion for each Time-Point
beta.disp.16.uni <- betadisper(prok.dist.uni.16hrs, bac_sperm_16@sam_data$Location_EC)
permutest(beta.disp.16.uni)
beta.disp.16.unib <- betadisper(prok.dist.uni.16hrs, bac_sperm_16@sam_data$Sample_Type)
permutest(beta.disp.16.unib)

```
###### Beta-Diversity using UnWeighted Unifrac Only Bacteria ####
```{r}
##### All Samples ########
# Create unweighted UniFrac distance matrix
bac.unifrac = UniFrac(BSpermosphere_RDS_CSS, weighted = FALSE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
adonis2(bac.unifrac ~ Location_EC * Sample_Type * Time, as(sample_data(BSpermosphere_RDS_CSS), "data.frame"))

# Perform ordination using MDS with unweighted UniFrac distance
global.ord.uni <- phyloseq::ordinate(BSpermosphere_RDS_CSS, "MDS", "unifrac", weighted = FALSE)

# Calculate percentage variation on axis one
variation.axis1 <- round(100 * global.ord.uni$values$Relative_eig[[1]], 2)

# Calculate percentage variation on axis two
variation.axis2 <- round(100 * global.ord.uni$values$Relative_eig[[2]], 2)

# Extract ordination vectors
bac.ggplot.data.uni <- data.frame(global.ord.uni$vectors)
bac.ggplot.data.uni$Sample_ID <- rownames(bac.ggplot.data.uni)

# Merge ordination vectors with sample metadata
bac.ggplot.data.uni2 <- left_join(data.frame(BSpermosphere_RDS_CSS@sam_data), bac.ggplot.data.uni, by = "Sample_ID")

# Order the time points in chronological order
bac.ggplot.data.uni2$Time.Point <- factor(bac.ggplot.data.uni2$Time, levels = c("0", "8", "12","16"))

# Create a ggplot for unweighted UniFrac distance
global.uni.unweighted <- ggplot(bac.ggplot.data.uni2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type, color = Location_EC)) +
  geom_point(size = 4, alpha = 0.7) +  # Add points with specified size and transparency
  scale_shape_manual(values = c(21, 22, 24, 25)) +  # Customize shapes for different types
  scale_fill_manual(values = cbbPalette) +  # Customize fill colors for different treatments
  scale_color_manual(values = cbbPalette) +  # Customize colors for different treatments
  xlab(paste("PcoA1 -", variation.axis1, "%")) +  # Label for x-axis with percentage variation 1
  ylab(paste("PcoA2 -", variation.axis2, "%")) +  # Label for y-axis with percentage variation 2
  theme(legend.position = "bottom", plot.title = element_text(size = 10, face = "bold")) +  # Customize theme
  ggtitle("Unweighted Unifrac")  # Add title to the plot
global.uni.unweighted  # Display the plot
```
### 0 hours; Unweighted Unifrac #####
```{r}
# Filter to time point
bac_sperm_0 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "0") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.0hrs = UniFrac(bac_sperm_0, weighted = FALSE)  # Create distance matrix
adonis.0.uni <- adonis2(prok.dist.uni.0hrs ~ Location_EC * Sample_Type, as(sample_data(bac_sperm_0), "data.frame"))  # Perform PERMANOVA to test for significant changes
adonis.0.uni  # Display PERMANOVA results

# The R2 is the variation due to different factors 
# pvalue is the p-value for the factor
pvalue <- adonis.0.uni$`Pr(>F)`[[1]]
R2 <- adonis.0.uni$R2[[1]]

# Plotting
ordination.unifrac.pcoa.0hrs <- ordinate(bac_sperm_0, "PCoA", distance = prok.dist.uni.0hrs)  # Calculate the resemblance and ordinate using PCoA
uni.0.values <- data.frame(ordination.unifrac.pcoa.0hrs$values)

# Percent variation
variation.axis1.0.uni <- round(100 * uni.0.values$Relative_eig[[1]], 2)
variation.axis2.0.uni <- round(100 * uni.0.values$Relative_eig[[2]], 2)

# Data from ordinations
uni.0.vectors <- data.frame(ordination.unifrac.pcoa.0hrs$vectors)  # Positions of your points on the PCoA graph
uni.0.vectors$Sample_ID <- rownames(uni.0.vectors)
uni.0.vectors2 <- left_join(data.frame(bac_sperm_0@sam_data), uni.0.vectors, by = "Sample_ID")

# Create a ggplot for unweighted UniFrac distance at 0 hours
un.pcoa.unifrac.0hrs.new <- ggplot(uni.0.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size = 4, alpha = 0.7) +
  scale_shape_manual(values = c(21, 24)) +
  scale_fill_manual(values = cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.0.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.0.uni, "%")) +
  guides(fill = guide_legend(override.aes = list(shape = 21))) +
  ggtitle(paste("Planting")) +
  theme_bw() +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.key = element_blank(),
        plot.title = element_text(size = 10, face = "bold"))
un.pcoa.unifrac.0hrs.new

# Dispersion for each Time-Point
beta.disp.0.uni <- betadisper(prok.dist.uni.0hrs, bac_sperm_0@sam_data$Location_EC)
permutest(beta.disp.0.uni)
beta.disp.0.uni.Ty <- betadisper(prok.dist.uni.0hrs, bac_sperm_0@sam_data$Sample_Type)

```
### 8 hours; Unweighted Unifrac #####
```{r}
bac_sperm_8 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "8") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.8hrs = UniFrac(bac_sperm_8, weighted = F) # create distance matrix
adonis.8.uni <- adonis2(prok.dist.uni.8hrs~Location_EC*Sample_Type, as(sample_data(bac_sperm_8), "data.frame")) #Are there significant changes?
adonis.8.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.8.uni$`Pr(>F)`[[1]]
R2 <- adonis.8.uni$R2[[1]]

# plotting
ordination.unifrac.pcoa.8hrs <- ordinate(bac_sperm_8, "PCoA", distance = prok.dist.uni.8hrs) # calculate the resemblance and ordinate using PCoA
uni.8.values <- data.frame(ordination.unifrac.pcoa.8hrs$values)

# percent variation
variation.axis1.8.uni <- round(100*uni.8.values$Relative_eig[[1]], 2)
variation.axis2.8.uni <- round(100*uni.8.values$Relative_eig[[2]], 2)

# data from ordinations
uni.8.vectors <- data.frame(ordination.unifrac.pcoa.8hrs$vectors) # positions of your points on the pcoa graph
uni.8.vectors$Sample_ID <- rownames(uni.8.vectors)
uni.8.vectors2 <- left_join(data.frame(bac_sperm_8@sam_data), uni.8.vectors,  by = "Sample_ID")
  
  un.pcoa.unifrac.8hrs.new <- ggplot(uni.8.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.8.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.8.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("8 Hours Postplanting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
un.pcoa.unifrac.8hrs.new

#Dispersion for each Time-Point
beta.disp.8.uni <- betadisper(prok.dist.uni.8hrs, bac_sperm_8@sam_data$Location_EC)
permutest(beta.disp.8.uni)
beta.disp.8.uni.b <- betadisper(prok.dist.uni.8hrs, bac_sperm_8@sam_data$Sample_Type)
permutest(beta.disp.8.uni.b)
```
### 12; Unweighted Unifrac #####
```{r}
bac_sperm_12 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "12") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.12hrs = UniFrac(bac_sperm_12, weighted = F) # create distance matrix
adonis.12.uni <- adonis2(prok.dist.uni.12hrs~Location_EC*Sample_Type, as(sample_data(bac_sperm_12), "data.frame")) #Are there significant changes?
adonis.12.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.12.uni$`Pr(>F)`[[1]]
R2 <- adonis.12.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.12hrs <- ordinate(bac_sperm_12, "PCoA", distance = prok.dist.uni.12hrs) # calculate the resemblance and ordinate using PCoA
uni.12.values <- data.frame(ordination.unifrac.pcoa.12hrs$values)

# percent variation
variation.axis1.12.uni <- round(100*uni.12.values$Relative_eig[[1]], 2)
variation.axis2.12.uni <- round(100*uni.12.values$Relative_eig[[2]], 2)

# data from ordinations
uni.12.vectors <- data.frame(ordination.unifrac.pcoa.12hrs$vectors) # positions of your points on the pcoa graph
uni.12.vectors$Sample_ID <- rownames(uni.12.vectors)
uni.12.vectors2 <- left_join(data.frame(bac_sperm_12@sam_data), uni.12.vectors,  by = "Sample_ID")

  un.pcoa.unifrac.12hrs.new <- ggplot(uni.12.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.12.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.12.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("12 Hours Postplanting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
un.pcoa.unifrac.12hrs.new

#Dispersion for each Time-Point
beta.disp.12.uni <- betadisper(prok.dist.uni.12hrs, bac_sperm_12@sam_data$Location_EC)
permutest(beta.disp.12.uni)
beta.disp.12.uni.b <- betadisper(prok.dist.uni.12hrs, bac_sperm_12@sam_data$Sample_Type)
permutest(beta.disp.12.uni.b)
```
### 16; Unweighted Unifrac #####
```{r}
bac_sperm_16 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "16") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.16hrs = UniFrac(bac_sperm_16, weighted = F) # create distance matrix
adonis.16.uni <- adonis2(prok.dist.uni.16hrs~Location_EC*Sample_Type, as(sample_data(bac_sperm_16), "data.frame")) #Are there significant changes?
adonis.16.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.16.uni$`Pr(>F)`[[1]]
R2 <- adonis.16.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.16hrs <- ordinate(bac_sperm_16, "PCoA", distance = prok.dist.uni.16hrs) # calculate the resemblance and ordinate using PCoA
uni.16.values <- data.frame(ordination.unifrac.pcoa.16hrs$values)

# percent variation
variation.axis1.16.uni <- round(100*uni.16.values$Relative_eig[[1]], 2)
variation.axis2.16.uni <- round(100*uni.16.values$Relative_eig[[2]], 2)

# data from ordinations
uni.16.vectors <- data.frame(ordination.unifrac.pcoa.16hrs$vectors) # positions of your points on the pcoa graph
uni.16.vectors$Sample_ID <- rownames(uni.16.vectors)
uni.16.vectors2 <- left_join(data.frame(bac_sperm_16@sam_data), uni.16.vectors,  by = "Sample_ID")

  un.pcoa.unifrac.16hrs.new <- ggplot(uni.16.vectors2, aes(x = Axis.1, y = Axis.2, fill = Location_EC, shape = Sample_Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.16.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.16.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("16 Hours Postplanting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
un.pcoa.unifrac.16hrs.new

#Dispersion for each Time-Point
beta.disp.16.uni <- betadisper(prok.dist.uni.16hrs, bac_sperm_16@sam_data$Location_EC)
permutest(beta.disp.16.uni)
beta.disp.16.uni.b <- betadisper(prok.dist.uni.16hrs, bac_sperm_16@sam_data$Sample_Type)
permutest(beta.disp.16.uni.b)
```
#combining beta diversity figures
```{r}

UnifracCombined <- ggpubr::ggarrange(pcoa.unifrac.0hrs.new,
                               pcoa.unifrac.8hrs.new,
                               pcoa.unifrac.12hrs.new,
                               pcoa.unifrac.16hrs.new,
                               un.pcoa.unifrac.0hrs.new,
                               un.pcoa.unifrac.8hrs.new,
                               un.pcoa.unifrac.12hrs.new,
                               un.pcoa.unifrac.16hrs.new,
                               labels = "auto",
                               nrow = 2, ncol = 4, common.legend = TRUE, legend = "bottom")
#A-D is Weighted Unifrac, E-H is unweighted Unifrac


```

#Steamed vs Nonsteamed Analysis#
#Relative Abundance#
##### Community Composition of top 20 most abundant OTUs ######
##### Prokaryote #####
```{r}

```
































